src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -202,22 +202,23 @@ private static IArgument DeriveArgument(int parameterIndex, int argumentIndex, I
 
         private static IOperation CreateParamArray(IParameterSymbol parameter, ImmutableArray<BoundExpression> boundArguments, int firstArgumentElementIndex, SyntaxNode invocationSyntax)
         {
-            if (parameter.Type.TypeKind == TypeKind.Array)
+            ArrayBuilder<IOperation> builder = ArrayBuilder<IOperation>.GetInstance(boundArguments.Length - firstArgumentElementIndex);
+            for (int index = firstArgumentElementIndex; index < boundArguments.Length; index++)
             {
-                IArrayTypeSymbol arrayType = (IArrayTypeSymbol)parameter.Type;
-                ArrayBuilder<IOperation> builder = ArrayBuilder<IOperation>.GetInstance(boundArguments.Length - firstArgumentElementIndex);
+                builder.Add(boundArguments[index]);
+            }
 
-                for (int index = firstArgumentElementIndex; index < boundArguments.Length; index++)
-                {
-                    builder.Add(boundArguments[index]);
-                }
-                var paramArrayArguments = builder.ToImmutableAndFree();
+            var paramArrayArguments = builder.ToImmutableAndFree();
 
+            if (parameter.Type.TypeKind == TypeKind.Array)
+            {
+                IArrayTypeSymbol arrayType = (IArrayTypeSymbol)parameter.Type;
+                
                 // Use the invocation syntax node if there is no actual syntax available for the argument (because the paramarray is empty.)
                 return new ArrayCreation(arrayType, paramArrayArguments, paramArrayArguments.Length > 0 ? paramArrayArguments[0].Syntax : invocationSyntax);
             }
 
-            return new InvalidExpression(invocationSyntax);
+            return new InvalidExpression(invocationSyntax, paramArrayArguments);
```

Comment:
>return new InvalidExpression(invocationSyntax, paramArrayArguments); [](start = 12, length = 68)

Though this looks like an improvement by comparison to the original code, it might not be the right thing to do to create an InvalidExpression for a scenario like this.